homeassistant/components/delijn/sensor.py

```diff
@@ -37,61 +38,64 @@
 async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
     """Create the sensor."""
     api_key = config[CONF_API_KEY]
-    name = DEFAULT_NAME
 
     session = async_get_clientsession(hass)
 
     sensors = []
     for nextpassage in config[CONF_NEXT_DEPARTURE]:
-        stop_id = nextpassage[CONF_STOP_ID]
-        number_of_departures = nextpassage[CONF_NUMBER_OF_DEPARTURES]
-        line = Passages(
-            hass.loop, stop_id, number_of_departures, api_key, session, True
+        sensors.append(
+            DeLijnPublicTransportSensor(
+                Passages(
+                    hass.loop,
+                    nextpassage[CONF_STOP_ID],
+                    nextpassage[CONF_NUMBER_OF_DEPARTURES],
+                    api_key,
+                    session,
+                    True,
+                )
+            )
         )
-        await line.get_passages()
-        if line.passages is None:
-            _LOGGER.warning("No data received from De Lijn")
-            return
-        sensors.append(DeLijnPublicTransportSensor(line, name))
 
     async_add_entities(sensors, True)
 
 
 class DeLijnPublicTransportSensor(Entity):
     """Representation of a Ruter sensor."""
 
-    def __init__(self, line, name):
+    def __init__(self, line):
         """Initialize the sensor."""
         self.line = line
         self._attributes = {ATTR_ATTRIBUTION: ATTRIBUTION}
-        self._name = name
+        self._name = None
         self._state = None
-        self._available = False
+        self._available = True
 
     async def async_update(self):
         """Get the latest data from the De Lijn API."""
-        await self.line.get_passages()
-        if self.line.passages is None:
-            _LOGGER.warning("No data received from De Lijn")
+        try:
+            await self.line.get_passages()
+            self._name = await self.line.get_stopname()
+        except HttpException:
+            self._available = False
+            _LOGGER.error("De Lijn http error")
             return
+
         try:
             first = self.line.passages[0]
             if first["due_at_realtime"] is not None:
                 first_passage = first["due_at_realtime"]
             else:
                 first_passage = first["due_at_schedule"]
             self._state = first_passage
-            self._name = first["stopname"]
-            self._attributes["stopname"] = first["stopname"]
```

Comment:
If we remove this attribute it will be a breaking change and we can't release this PR in a bug fix release. I suggest we duplicate the information for now and set the `"stopname"` attribute from `self._name`.

We can remove the duplicate attribute in a separate PR.